src/Compilers/VisualBasic/Portable/CodeGen/ResumableStateMachineStateAllocator.vb

```diff
@@ -0,0 +1,91 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports System
+Imports System.Diagnostics
+Imports Microsoft.CodeAnalysis.CodeGen
+Imports Microsoft.CodeAnalysis.VisualBasic.Symbols
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic
+    ''' <summary>
+    ''' Allocates resumable states, i.e. states that resume execution of the state machine after await expression or yield return.
+    ''' </summary>
+    Friend NotInheritable Class ResumableStateMachineStateAllocator
```

Comment:
This type seems identical to `Microsoft.CodeAnalysis.CSharp.ResumableStateMachineStateAllocator`. Can we share it?

Reply:
Unfortunately not since bound nodes are language specific.

Reply:
Yeah, I thought about a base type but end the end of the day the amount of shared code is so small that it doesn't seem worth it.